### PR TITLE
fix: Deduplicated all test id props with data-testid

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR removes all the data-cy, data-test, cypressSelector, and data-test-id. instead implemented data-testid.

#### Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update

#### How Has This Been Tested?

- [ ] Manual
- [ ] Jest
- [ ] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/cypress/support/Pages/HomePage.ts [hunk 1]
```diff
@@ -172,12 +170,14 @@ export class HomePage {
     cy.xpath(this._selectRole).first().click({ force: true });
     this.agHelper.Sleep(500);
     cy.xpath(this._userRole(role)).click({ force: true });
-    this.agHelper.ClickButton("Invite");
+    this.agHelper.GetNClick(this._inviteButton, 0, true);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/HomePage.ts:173]: Can we use - ` this.agHelper.ClickButton("Invite"); `itself here? I see you have edited the _spanButton accordingly in Commonlocators.

File: app/client/cypress/support/Pages/AggregateHelper.ts [hunk 2]
```diff
@@ -661,6 +661,17 @@ export class AggregateHelper {
     }
   }
 
+  public AssertSelectedTab(propertyName: string, value: "true" | "false") {
+    let locator;
+    if (propertyName.startsWith("//")) {
+      locator = cy.xpath(propertyName);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/AggregateHelper.ts:667]: agHelper.GetElement can be used.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Templates/Fork_Template_spec.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ describe("excludeForAirgap", "Fork a template to an workspace", () => {
     );
   });
 
-  it("4. Check if tooltip is working in 'Reconnect Datasources'", () => {
+  it.only("4. Check if tooltip is working in 'Reconnect Datasources'", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Templates/Fork_Template_spec.js:61]: Only here is intentional?

Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Templates/Fork_Template_spec.js:61]: Removed!

Comment (Parthvi12) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Templates/Fork_Template_spec.js]: Oops no

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Widgets/List/List4_spec.js [hunk 4]
```diff
@@ -67,7 +67,7 @@ describe("Container Widget Functionality", function () {
     _.entityExplorer.ExpandCollapseEntity("Widgets");
     _.entityExplorer.ExpandCollapseEntity("List1");
     _.entityExplorer.ExpandCollapseEntity("Container1");
-    _.entityExplorer.SelectEntityByName("List1");
+    _.entityExplorer.SelectEntityByName("Text1");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Widgets/List/List4_spec.js:70]: Why its changed to Text1? the spec is List related know?

Comment (albinAppsmith) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Widgets/List/List4_spec.js:70]: This was Text1 before and you changed it seems.. then tests were failing

Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Widgets/List/List4_spec.js:70]: Oh got it,, may be part of SelectEntityByName change, it got updated. ThankYou.

File: app/client/cypress/e2e/Regression_TestSuite/ClientSideTests/Binding/Bind_InputWidget_TableV2_Sorting_spec.js [hunk 5]
```diff
@@ -19,7 +21,8 @@ describe("Binding the Table and input Widget", function () {
   });
 
   it("2. validation of data displayed in input widgets based on sorting", function () {
-    cy.SearchEntityandOpen("Table1");
+    _.entityExplorer.SelectEntityByName("Table1");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression_TestSuite/ClientSideTests/Binding/Bind_InputWidget_TableV2_Sorting_spec.js:24]: Hello! trial comment

File: app/client/cypress/support/Pages/HomePage.ts [hunk 6]
```diff
@@ -171,12 +170,14 @@ export class HomePage {
...
     cy.wait("@mockPostInvite")
       .its("request.headers")
       .should("have.property", "origin", "Cypress");
-    cy.contains(email, { matchCase: false });
-    cy.contains(successMessage);
+    // cy.contains(email, { matchCase: false });
+    //this.agHelper.ValidateToastMessage(   commenting until method is fixed
+    //  successMessage
+    // );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/HomePage.ts:180]: To check - why comment here.

File: app/client/cypress/support/Pages/HomePage.ts [hunk 7]
```diff
@@ -427,11 +427,11 @@ export class HomePage {
     cy.xpath(this._selectRole).first().click({ force: true });
     this.agHelper.Sleep(500);
     cy.xpath(this._userRole(role)).click({ force: true });
-    this.agHelper.ClickButton("Invite");
+    this.agHelper.GetNClick(this._inviteButton, 0, true);
     cy.wait("@mockPostInvite")
       .its("request.headers")
       .should("have.property", "origin", "Cypress");
-    cy.contains(email, { matchCase: false });
+    // cy.contains(email, { matchCase: false });
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/HomePage.ts:434]: Check why are these commented.

Comment (Parthvi12) [reply app/client/cypress/support/Pages/HomePage.ts:434]: The share modal doesnt show email anymore, hence commented the assertion for that.

File: app/client/cypress/support/Pages/HomePage.ts (hunk 6 above)
Comment (Parthvi12) [reply app/client/cypress/support/Pages/HomePage.ts]: Commented it as toast message was failing , added back.

**Pull request merged**